src/Core/src/Core.csproj

```diff
@@ -5,7 +5,7 @@
     <RootNamespace>Microsoft.Maui</RootNamespace>
     <AssemblyName>Microsoft.Maui</AssemblyName>
     <Nullable>enable</Nullable>
-    <IsTrimmable>false</IsTrimmable>
+    <IsAotCompatible Condition="'$(TargetFramework)' != 'netstandard2.0' and '$(TargetFramework)' != 'netstandard2.1'">true</IsAotCompatible>
```

Comment:
This is problematic, this adds the `IsTrimmable=true` assembly attribute.
* I have an app for .NET 8 which does reflection on something in `Core` - the reflection is from the app's code or from some control, or binding...
* In .NET 8 `Core` is not trimmed and so everything works.
* I migrate the app to .NET 9, where `Core` would now be trimmed (due to the new attribute).
* My app breaks on 9, because the thing I reflect on from `Core` is removed by the trimmer - in the default MAUI Release configuration (No NativeAOT anywhere).

I think we should enable the analyzers, but we should have a discussion if adding `IsTrimmable=true` is something we want to do for 9.

Reply:
Good point, I'll enable just the analyzers for now.

Reply:
Yes, `IsTrimmable=true` would make the assembly trimmed if `TrimMode=partial`, which is the current default.

There are a couple assemblies in here that already have `IsTrimmable=true`: Essentials, Compatibility, maybe even Graphics? Those could probably enable IsAotCompatible without issue.